src/core/Stylesheet.ts

```diff
@@ -5,6 +5,8 @@ interface IRule {
     selectorText: string;
 }
 
+export const sanitizeId = (id: string) => id.replace(/([`~\!\@#\$%\^&\*\(\)=\+\\|\/,.'\";:\?><\[\]\{\}\s])/g, '\\$1');
```

Comment:
There's a little more to it unfortunately... at least leading digits (`id="123"`) and given the complexity of discussions about this I suspect there's more. Maybe just pull in https://github.com/mathiasbynens/CSS.escape unless we already have `CSS.escape` in our polyfills?

Reply:
I was trying to deal with the most common offenders, but you're right, might as well just deal with all cases. `CSS.escape` is not included in the polyfill library we use. Since only the table requires it, I'll include it to the table and add tests for 2 typical cases (leading number, leading underscore not followed by a letter)